sw/device/tests/alert_handler_reverse_ping_in_deep_sleep_test.c

```diff
@@ -0,0 +1,229 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/math.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "alert_handler_regs.h"  // Generated.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  // This time needs to be greater than 0.175s. See test plan for more details.
+  kTestParamWakeupThresholdUsec = 200000,
+  kTestParamAlertHandlerIrqDeadlineUsec = 100,
+  kTestParamAlertHandlerPhase0EscalationDurationUsec = 100,
+  kTestParamAlertHandlerPingTimeoutUsec = 20,
+};
+
+static_assert(
+    kTestParamWakeupThresholdUsec > 175000,
+    "Invalid kTestParamWakeupThresholdUsec. See test plan for more details.");
+
+static dif_rv_plic_t plic;
+static dif_pwrmgr_t pwrmgr;
+static dif_rstmgr_t rstmgr;
+static dif_aon_timer_t aon_timer;
+static dif_alert_handler_t alert_handler;
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+
+static plic_isr_ctx_t plic_ctx = {
+    .rv_plic = &plic,
+    .hart_id = kPlicTarget,
+};
+
+static alert_handler_isr_ctx_t alert_handler_ctx = {
```

Comment:
It is used in the ISR to ack any interrupts that may fire, but I can remove it. It should not cause any side effects.

Reply:
Sure, I think it would be nice to remove it. Because this one expect to see an interrupt from class B, but in this test, we do not expect to see any interrupt :)